### PR TITLE
fix issue when BCPROP is not set initially

### PR DESCRIPTION
**Pull request by totto82**
See https://github.com/OPM/opm-models/pull/833

### CODE REVIEW

File: ebos/eclproblem.hh [hunk 1]
```diff
@@ -1193,8 +1193,10 @@ public:
                 values.setThermalFlow(context, spaceIdx, timeIdx, boundaryFluidState(globalDofIdx, indexInInside));
             else if (type == BCType::FREE || type == BCType::DIRICHLET)
                 values.setFreeFlow(context, spaceIdx, timeIdx, boundaryFluidState(globalDofIdx, indexInInside));
-            else
+            else if (type == BCType::RATE)
                 values.setMassRate(massrate, pvtRegionIdx);
+            else
```
Comment (akva2) [inline ebos/eclproblem.hh:1198]: won't this overwrite the thermal condition from the if above this nest?

Comment (totto82) [reply ebos/eclproblem.hh:1198]: I dont think we use enableThermalFluxBoundaries much but youre right. I will fix. Thanks for noticing.

File: ebos/eclproblem.hh [hunk 2]
```diff
@@ -1459,103 +1459,105 @@ public:
...
-            std::array<Scalar, numPhases> pc = {0};
-            const auto& matParams = materialLawParams(globalDofIdx);
-            MaterialLaw::capillaryPressures(pc, matParams, fluidState);
-            Valgrind::CheckDefined(pressure);
-            Valgrind::CheckDefined(pc);
-            for (unsigned phaseIdx = 0; phaseIdx < numPhases; ++phaseIdx) {
-                if (!FluidSystem::phaseIsActive(phaseIdx))
-                    continue;
+        const auto& bcprop = this->simulator().vanguard().schedule()[this->episodeIndex()].bcprop;
+        if (bcprop.size() > 0){
```
Comment (atgeirr) [inline ebos/eclproblem.hh:1463]: If I understand correctly, unless we have a dirichlet condition we just return the initialFluidStates_[globalDofIdx], and that is not changed by this PR. The new if her is just to avoid indexing into an empty bcprop?

Nitpick: space before {

File: ebos/eclproblem.hh [hunk 3]
```diff
@@ -1459,103 +1459,105 @@ public:
     const InitialFluidState boundaryFluidState(unsigned globalDofIdx, const int directionId) const
     {
         OPM_TIMEBLOCK_LOCAL(boundaryFluidState);
-        FaceDir::DirEnum dir = FaceDir::FromIntersectionIndex(directionId);
-        assert(bcindex_(dir)[globalDofIdx] > 0);
```
Comment (atgeirr) [inline ebos/eclproblem.hh:1463]: Why remove the assert()? What are the requirements/range for this index?

**Review by atgeirr — COMMENTED** [review]
Looks like a very minor change to me, that should be merged.

File: ebos/eclproblem.hh (hunk 3 above)
Comment (totto82) [reply ebos/eclproblem.hh:1463]: Thanks for noticing. I just removed it during debugging and forgot to add it back.

**Review by atgeirr — APPROVED** [review]

**Comment by totto82** [comment]
jenkins build this opm-models=833 update_data please

File: ebos/eclproblem.hh (hunk 3 above)
Comment (totto82) [reply ebos/eclproblem.hh:1463]: Seems like adding the assert() triggered a test failure. I will investigate.

Comment (totto82) [reply ebos/eclproblem.hh:1463]: Since we now ask for the boundaryState at every boundary face (also the ones where no boundary conditions are set) we can not use assert here. I instead added a explicit check and return to avoid accessing the bcprops for  invalid indices.
